src/blocks/shared/utils/date.ts

```diff
@@ -1,7 +1,33 @@
-export const SHORT_DATE_FORMAT: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric', year: 'numeric' };
-export const SHORT_DATE_FORMAT_WITHOUT_YEAR: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' };
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+const SHORT_DATE_FORMAT: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric', year: 'numeric' };
+const SHORT_DATE_FORMAT_WITHOUT_YEAR: Intl.DateTimeFormatOptions = { month: 'short', day: 'numeric' };
+const DATE_NOT_AVAILABLE_MESSAGE = __( 'Date N/A', 'wp-parsely' );
+
+/**
+ * Returns whether the passed date can be processed further.
+ *
+ * @param {Date} date The date to be examined.
+ * @return {boolean} Whether the date can be processed further.
+ */
+export function canProcessDate( date: Date|string ): boolean {
+	if ( 'string' === typeof date ) {
+		date = new Date( date );
+	}
+
+	// Return false if the object is not a valid Date object, or if its value is
+	// equal to the Unix Epoch.
+	return date instanceof Date && ! isNaN( +date ) && 0 !== date.getTime();
+}
 
 export function getDateInUserLang( date: Date, options: Intl.DateTimeFormatOptions ): string {
```

Comment:
With the newest changes, we aren't using this function but I have not removed it, in case we want to display dates differently in the future.